features/eolearn/features/clustering.py

```diff
@@ -0,0 +1,145 @@
+"""
+Module for computing clusters in EOPatch
+"""
+
+import numpy as np
+from sklearn.cluster import AgglomerativeClustering
+from sklearn.feature_extraction.image import grid_to_graph
+from eolearn.core import EOTask, FeatureType
+
+
+class ClusteringTask(EOTask):
+    """
+    Tasks computes clusters on selected features using sklearn.cluster.AgglomerativeClustering.
+
+    The algorithm produces a timeless data feature where each cell has a natural number which corresponds to specific
+    group. The cells marked with -1 are not marking clusters. They are either being excluded by a mask or later removed
+    by depending on the 'remove_small' threshold.
+    """
```

Comment:
Explicitly mention in the documentation that input features need to be timeless.